src/Resharper.ConfigurationSense/Extensions/ProjectExtensions.cs

```diff
@@ -26,7 +27,7 @@ public static class ProjectExtensions
         {
             var configFiles = GetNetCoreJsonConfigFiles(project);
 
-            var settings = new HashSet<KeyValueSetting>(KeyValueSetting.KeyComparer);
+            var settings = new Dictionary<string, IList<string>>();
```

Comment:
Let's use linked list here
```suggestion
            var settings = new Dictionary<string, LinkedList<string>>();
```